framework/src/actions/SetupMeshAction.C

```diff
@@ -152,6 +153,34 @@ SetupMeshAction::setupMesh(MooseMesh * mesh)
 void
 SetupMeshAction::act()
 {
+  // switch non-file meshes to be a file-mesh if using a pre-split mesh configuration.
+  if (_app.parameters().get<bool>("use_split"))
+  {
+    auto split_file = _app.parameters().get<std::string>("split_file");
+    if (_type != "FileMesh")
+    {
+      if (split_file == "")
+        mooseError("cannot use split mesh with non-file mesh without specifying --split-file on "
+                   "command line");
```

Comment:
I guess this is OK, it would be cool if we could use the virtual function but that might be a little tricky *before* the object is created.

Reply:
For generated mesh, we don't know the name of the pre-split files.  There is no way we could divine it.  The autosplit generated mesh and run in one go is a separate thing that will have to be implemented later.

Reply:
I'm with @rwcarlsen here - this is the right thing to do.